agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -329,6 +368,11 @@ def labels_outdated?(service_container)
       service_pod.labels['io.kontena.load_balancer.name'] != service_container.labels['io.kontena.load_balancer.name']
     end
 
+    # @return [Boolean]
+    def restart_service_container?(service_container)
+      @restarting_at && service_container.started_at <= @restarting_at
```

Comment:
This is comparing a timestamp provided by the Docker engine with a `Time.now` timestamp determined by the agent... I wouldn't expect those to ever differ, but this could probably be avoided...